### PR TITLE
Handle nested non-deconstruction conversions in DeconstructionInfo

### PR DESCRIPTION
**Pull request by 333fred**
If there were nested non-deconstruction conversions in a deconstruction, then we'd blindly recurse. Previous to https://github.com/dotnet/roslyn/pull/57175, we'd have non-terminal conversion nodes in the info tree, which was equally broken, though it didn't crash, just returned potentially incorrect results. Fixes https://github.com/dotnet/roslyn/issues/61332.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs:6554]:
```suggestion
            Assert.All(nestedConversions, n => Assert.Empty(n.Nested));
```

**Comment by 333fred** [comment]
https://github.com/orgs/dotnet/teams/roslyn-compiler for review.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for reviews.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for reviews please.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review please.

File: src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/Semantics/DeconstructionTests.cs:6535]:
```suggestion
        [Fact, WorkItem(61332, "https://github.com/dotnet/roslyn/issues/61332")]
```

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3), assuming CI is passing

**Pull request merged**
